src/main/charts/bitbucket/templates/_helpers.tpl

```diff
@@ -102,10 +102,18 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
 
 {{- define "bitbucket.baseUrl" -}}
-{{ $httpPortMismatch := (and (eq .Values.ingress.scheme "http") (ne (int .Values.ingress.port) 80) ) -}}
-{{ $httpsPortMismatch := (and (eq .Values.ingress.scheme "https") (ne (int .Values.ingress.port) 443) ) -}}
-{{ .Values.ingress.scheme }}://{{ .Values.ingress.host -}}
-{{ if or $httpPortMismatch $httpsPortMismatch }}:{{ .Values.ingress.port }}{{ end }}
+{{ ternary "https" "http" .Values.ingress.https -}}
+://
+{{- .Values.ingress.host -}}
+{{ with .Values.ingress.port }}:{{ . }}{{ end }}
+{{- end }}
```

Comment:
I've rewritten the above function to be a little less terse. It'll include the `port` in the baseUrl if (and only if) it's specified in values.